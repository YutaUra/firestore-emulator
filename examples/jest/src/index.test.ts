import { Firestore } from "@google-cloud/firestore";

describe("can create document", () => {
  process.env["GOOGLE_CLOUD_PROJECT"] = "test-project";
  process.env["FIRESTORE_EMULATOR_HOST"] = emulator.host;
  const db = new Firestore();

  it("can create document", async () => {
    emulator.state.emitter.on("create-document", ({ document }) => {
      console.log("create-document", document);
    });
    await db.collection("users").doc("alice").set({
      name: "Alice",
    });

    const doc = await db.collection("users").doc("alice").get();

    expect(doc.exists).toBe(true);
    expect(doc.data()).toEqual({
      name: "Alice",
    });
    console.log(emulator.state.toJSON());
    expect(emulator.state.toJSON()).toMatchSnapshot();

    emulator.state.emitter.removeAllListeners();
  });
});
