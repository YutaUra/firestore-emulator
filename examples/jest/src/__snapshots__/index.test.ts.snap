// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can create document can create document 1`] = `
{
  "projects": {
    "test-project": {
      "databases": {
        "(default)": {
          "collections": {
            "users": {
              "documents": {
                "alice": {
                  "collections": {},
                  "fields": {
                    "name": "Alice",
                  },
                  "path": "projects/test-project/databases/(default)/users/alice",
                },
              },
              "path": "projects/test-project/databases/(default)/users",
            },
          },
          "path": "projects/test-project/databases/(default)",
        },
      },
      "path": "projects/test-project",
    },
  },
}
`;
